src/Lab.java

```diff
@@ -19,22 +26,42 @@ public static void main(String[] args) throws IOException{
         DatagramPacket r = new DatagramPacket(new byte[UDP_SIZE], UDP_SIZE);
         DatagramSocket s = new DatagramSocket(LISTEN_PORT);
         s.receive(r);
-        InetAddress userAddress = r.getAddress();
+        // save port & address for future response
         int userPort = r.getPort();
-        // Send packet to root
-        DatagramSocket senderSocket = new DatagramSocket();
-        byte[] data = Arrays.copyOf(r.getData(), r.getLength());
-        DatagramPacket query = new DatagramPacket(data, r.getLength(),getRoot(),DNS_PORT);
-        senderSocket.send(query);
-        // Get response from root
-        senderSocket.receive(r);
-        data = Arrays.copyOf(r.getData(), r.getLength());
-        // Query name servers iteratively
-        while (isNoError(data) && getAnswer(data) == 0 && getAuthority(data) > 0){
-            InetAddress address = getNextServer(data);
-            senderSocket.send(new DatagramPacket(query.getData(),query.getLength(), address, DNS_PORT));
+        InetAddress userAddress = r.getAddress();
+
+        // get domain's name
+        String domainName = "";
+        byte[] data = Arrays.copyOfRange(r.getData(), 13, r.getLength());
+        for (byte cell : data) {
+            if (cell == 0) break;
+            if (cell < 20)
```

Comment:
through debugging all '.' signs ascii where below 10. I put that number only by improvising for the correctness of the check. A better solution will be replace this.